plugins/account_history/plugin.cpp

```diff
@@ -135,242 +157,290 @@ if( options.count(name) ) { \
-            impacted.insert(op.from);
-            impacted.insert(op.to);
+            if (op.from == op.to) {
+                impacted.insert(make_pair(op.from, operation_direction_type::any));
+                impacted.insert(make_pair(op.to, operation_direction_type::any));
+            } else {
+                impacted.insert(make_pair(op.from, operation_direction_type::sender));
+                impacted.insert(make_pair(op.to, operation_direction_type::receiver));
+            }
         }
 
         void operator()(const cancel_transfer_from_savings_operation& op) {
-            impacted.insert(op.from);
+            impacted.insert(make_pair(op.from, operation_direction_type::any));
         }
 
         void operator()(const decline_voting_rights_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::any));
         }
 
         void operator()(const comment_benefactor_reward_operation& op) {
-            impacted.insert(op.benefactor);
-            impacted.insert(op.author);
+            impacted.insert(make_pair(op.benefactor, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.author, operation_direction_type::sender));
         }
 
         void operator()(const delegate_vesting_shares_operation& op) {
-            impacted.insert(op.delegator);
-            impacted.insert(op.delegatee);
+            impacted.insert(make_pair(op.delegator, operation_direction_type::sender));
+            impacted.insert(make_pair(op.delegatee, operation_direction_type::receiver));
         }
 
         void operator()(const return_vesting_delegation_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::receiver));
         }
 
         void operator()(const proposal_create_operation& op) {
-            impacted.insert(op.author);
+            impacted.insert(make_pair(op.author, operation_direction_type::any));
         }
 
         void operator()(const proposal_update_operation& op) {
-            impacted.insert(op.active_approvals_to_add.begin(), op.active_approvals_to_add.end());
-            impacted.insert(op.owner_approvals_to_add.begin(), op.owner_approvals_to_add.end());
-            impacted.insert(op.posting_approvals_to_add.begin(), op.posting_approvals_to_add.end());
-            impacted.insert(op.active_approvals_to_remove.begin(), op.active_approvals_to_remove.end());
-            impacted.insert(op.owner_approvals_to_remove.begin(), op.owner_approvals_to_remove.end());
-            impacted.insert(op.posting_approvals_to_remove.begin(), op.posting_approvals_to_remove.end());
+            auto insert_fn = [this](const fc::flat_set<golos::chain::account_name_type>& impd) {
+                for (auto i : impd) {
+                    impacted.insert(make_pair(i, operation_direction_type::any));
```

Comment:
why not `sender` ?

Reply:
Other sides do not take part of this operation, so they will not be marked as receivers. The only other actor of this operation (excluding initiator) is `proposal.author`.
For example, if we have proposal from Dave, containing operations by Alice and Bob:
1. Dave is sender on proposal creation. Alice and Bob can be receivers, but it can be complicated, so ignore.
2. On proposal update only Alice or Bob can add their approvals. Normally, there will be 2 operations, one from Alice and other from Bob. Each will have one sender — account who adds/removes approval. Proposal author can be receiver. Other required accounts can be receivers too, but operation have no info about them.

Or did you mean something else?

Reply:
Actually, `sender` and `receiver` are not perfect terms here. `active` and `passive` or maybe `impactor`/`impacted` can be more clear, but still not perfect with `rewards` and virtual ops. So I agree on any case, we always can fix it later if better decision found.